tutorial/tests/testsuite.py

```diff
@@ -1,14 +1,43 @@
 import io
+import pathlib
 from contextlib import redirect_stdout
 from typing import Callable
 
 import pytest
-import ipynbname
 from IPython.core.interactiveshell import InteractiveShell
 from IPython.core.magic import Magics, magics_class, cell_magic
 from IPython.display import HTML, display
 
 
+def _name_from_line(line: str = None):
+    return line.strip().removesuffix(".py") if line else None
+
+
+def _name_from_ipynbname() -> str | None:
+    try:
+        import ipynbname
```

Comment:
Maybe it is better to keep the import at its original place. The module is still present, we are not removing it from the dependencies.